models/base/manifest/snowplow_media_player_base_quarantined_sessions.sql

```diff
@@ -0,0 +1,29 @@
+{{
+  config(
+    materialized='incremental',
+    full_refresh=snowplow_media_player.allow_refresh(),
+    sql_header=snowplow_utils.set_query_tag(var('snowplow__query_tag', 'snowplow_dbt')),
+    tblproperties={
+      'delta.autoOptimize.optimizeWrite' : 'true',
+      'delta.autoOptimize.autoCompact' : 'true'
+    }
+  )
+}}
+
+/*
+Boilerplate to generate table.
+Table updated as part of post-hook on sessions_this_run
+Any sessions exceeding max_session_days are quarantined
+Once quarantined, any subsequent events from the session will not be processed.
+This significantly reduces table scans
+*/
+
+with prep as (
+  select
+    cast(null as {{ dbt.type_string() }}) as session_id
+)
```

Comment:
For complicated reasons that come down to "dbt sucks at supporting warehouses properly" this should be ` cast(null as {% if target.type == 'redshift' %} varchar(400) {% else %} {{snowplow_utils.type_max_string()}} {% endif %}) session_id`